openmdao/surrogate_models/kriging.py

```diff
@@ -150,6 +156,14 @@ def _calculate_reduced_likelihood_params(self, thetas=None):
         params['U'] = U
         params['Vh'] = Vh
 
+        # Newly added for EGOLF
+        one = np.ones([self.n_samples,1])
+        R_inv = Vh.T.dot(np.dot(np.diag(inv_factors),U.T)) #Use einsum in the future release for efficiency
```

Comment:
Is R^{-1} itself needed or just the ability to multiply it against a vector (e.g. `lambda x: Vh.T.dot(np.einsum('j,kj,kl->jl', inv_factors, U, x))`)? If the latter, no need to calculate/store another full matrix.